zppy/templates/slurm_header.sh

```diff
@@ -11,9 +11,9 @@
 #SBATCH  --qos={{ qos }}
 #SBATCH  --partition={{ partition }}
 
-{% elif machine == 'cori' %}
+{% elif machine in ['pm-cpu', 'pm-gpu'] %}
```

Comment:
It should be possible to have if for the particular template arguments, rather than the machines so things could be much more compact.  See what we do in compass:
https://github.com/MPAS-Dev/compass/blob/master/compass/job/job_script.template

Here's how we call it:
https://github.com/MPAS-Dev/compass/blob/master/compass/job/__init__.py#L87-L90

Reply:
The result has a bit too much whitespace, so we clean it up:
https://github.com/MPAS-Dev/compass/blob/master/compass/job/__init__.py#L101-L122

Reply:
Since this isn't required for Perlmutter support, I created #333 to address this.